app/models/concerns/hud_concerns/client.rb

```diff
@@ -45,5 +45,25 @@ def gender_multi
         gm << self.GenderNone if gm.empty? && self.GenderNone.in?([8, 9, 99])
       end
     end
+
+    scope :age_group, ->(start_age: 0, end_age: nil) do
+      start_age = 0 unless start_age.is_a?(Integer)
+      end_age   = nil unless end_age.is_a?(Integer)
+      if end_age.present?
+        where(DOB: end_age.years.ago..start_age.years.ago)
+      else
+        where(arel_table[:DOB].lteq(start_age.years.ago))
+      end
+    end
+
+    scope :age_group_within_range, ->(start_age: 0, end_age: nil, start_date: Date.current, end_date: Date.current) do
+      start_age = 0 unless start_age.is_a?(Integer)
+      end_age   = nil unless end_age.is_a?(Integer)
+      if end_age.present?
+        where(DOB: (start_date - end_age.years)..(end_date - start_age.years))
+      else
+        where(arel_table[:DOB].lteq(start_date - start_age.years))
```

Comment:
should this be `end_date`?  Also see above about open ranges.

Reply:
I don't know honestly I copied these from the warehouse code https://github.com/greenriver/hmis-warehouse/pull/3123/commits/8704b10895643a4fe43d7341e3b46cd96f4e4869 to replace what @ZachCMP had written, just because I don't think we should have duplicate code for this.

Sorry @ZachCMP I was rushing through code review and making updates myself, since I had kept it sitting for several days already.

His code was 
```
  scope :with_age_range, ->(range) do
    query = c_t[:DOB].lteq(Date.today.years_ago(range.begin))
    query = query.and(c_t[:DOB].gteq(Date.today.years_ago(range.end))) unless range.end == Float::INFINITY

    where(query)
  end

```

Reply:
Yeah, I'm bad at date/age math.  What's there is what was in Client, so we should probably leave it for now.

Reply:
Anyway, sorry for wall of text. To address your question, I don't think so.
> should this be end_date? Also see above about open ranges.

This line of code you commented on is an "else" bock where there is no end date.

Reply:
sorry, just trying to determine if 
```
where(arel_table[:DOB].lteq(start_date - start_age.years))
```
should be:
```
where(arel_table[:DOB].lteq(end_date - start_age.years))
```
to match the `if` clause.

Reply:
`end_date` is nil, though. Unless I'm really misunderstanding..

Reply:
end_age is nil, but I'm pretty sure we still have an end date to the date range.  Anyway, let's leave it as is for now, since it's a straight port from the old code, which presumably is working.

Reply:
ohh 🤦 
okay, yeah presumably it works correctly..